### PR TITLE
Deprecate Fine Grain Settings for Scripts

### PR DESCRIPTION
**Pull request by jdconrad**
As the title says.  Will be replaced with the settings defined in #24532 in the next PR.

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/script/ScriptModesTests.java [hunk 1]
```diff
@@ -94,6 +97,21 @@ public void assertAllSettingsWereChecked() {
         }
     }
 
+    private static Setting<?>[] buildDeprecatedSettingsArray(ScriptSettings scriptSettings, String... keys) {
```
Comment (rjernst) [inline core/src/test/java/org/elasticsearch/script/ScriptModesTests.java]: This seems to be common, maybe just put it under one test class as public static?

**Review by rjernst — CHANGES_REQUESTED** [review]
Looks good, but I think this is still missing deprecation on the `langAndTypeSetting` in ScriptSettings?

File: core/src/test/java/org/elasticsearch/script/ScriptModesTests.java (hunk 1 above)
Comment (jdconrad) [inline core/src/test/java/org/elasticsearch/script/ScriptModesTests.java]: Do you have a class you prefer considering all of these inherit from ESTestCase which didn't seem appropriate?

**Comment by jdconrad** [comment]
@rjernst Thought langAndTypeSettng was done at line 110.  Did I miss a different place?

**Comment by rjernst** [comment]
What about line 94?

File: core/src/test/java/org/elasticsearch/script/ScriptModesTests.java (hunk 1 above)
Comment (jdconrad) [inline core/src/test/java/org/elasticsearch/script/ScriptModesTests.java]: Moved to ScriptSettingsTests

**Comment by jdconrad** [comment]
Think I found them all.

**Review by rjernst — APPROVED** [review]
Thanks! LGTM

**Comment by jdconrad** [comment]
@elasticmachine please test this

**Comment by jdconrad** [comment]
@rjernst Thanks for the review!

**Pull request merged**